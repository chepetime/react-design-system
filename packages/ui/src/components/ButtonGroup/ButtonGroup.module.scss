@use "./../../styles" as styles;

:root {
  --buttongroup-background-color: #{styles.$pink-500};
}

:global .dark {
  --buttongroup-background-color: #{styles.$indigo-500};
}

.ButtonGroup {
  @include styles.styles-reset;
  @include styles.text-wrapping;

  background: var(--buttongroup-background-color);

  font-size: 1rem;

  @include styles.respond-to("md") {
    font-size: 1.5rem;
  }
  @include styles.respond-to("lg") {
    font-size: 2rem;
  }
  @include styles.feature-to("hover") {
    &:hover {
      background-color: pink;
    }
  }
}
