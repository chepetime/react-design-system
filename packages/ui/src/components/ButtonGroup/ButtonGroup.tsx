import React from "react";
import classnames from "classnames";
import css from "./ButtonGroup.module.scss";

export interface ButtonGroupProps extends React.ComponentPropsWithRef<"div"> {
  children?: React.ReactNode;
}

export const ButtonGroup = React.forwardRef<HTMLDivElement, ButtonGroupProps>(
  (props: ButtonGroupProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.ButtonGroup)} ref={ref}>
        <p>ButtonGroup component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

ButtonGroup.displayName = "ButtonGroup";
