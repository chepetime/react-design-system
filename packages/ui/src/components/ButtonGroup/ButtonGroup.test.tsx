import React from "react";import { render } from "@testing-library/react";
import { ButtonGroup } from "./ButtonGroup";

describe("ButtonGroup", () => {
  test("renders the ButtonGroup component", () => {
    render(<ButtonGroup />);
  });
});
