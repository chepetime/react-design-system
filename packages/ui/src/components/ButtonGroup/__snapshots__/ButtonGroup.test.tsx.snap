// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup matches snapshot 1`] = `
<div>
  <div
    class="ButtonGroup"
  >
    <p>
      ButtonGroup component working!
    </p>
    <div />
  </div>
</div>
`;
